Small improvements

pblasucci opened the pull request:
- Added `warn` combinator
- Renamed `Combinators` module to `Trial`
- Added several missing XmlDoc comments

pblasucci committed:
Small improvements

+ Added `warn` combinator
+ Renamed `Combinators` module to `Trial`
+ Added several missing XmlDoc comments

forki commented:
I'm on vacation, but feel free to :shipit: and release it.

forki commented on src/Chessie/ErrorHandling.fs:
```diff
@@ -3,11 +3,10 @@ namespace Chessie.ErrorHandling
 type Result<'TSuccess, 'TMessage> = 
     /// Represents the result of a successful computation.
-    | Ok of 'TSuccess * 'TMessage list
+    | Okay of 'TSuccess * 'TMessage list
```
Not a big fan of this.

forki commented on src/Chessie/ErrorHandling.fs:
```diff
@@ -18,25 +17,28 @@ type Result<'TSuccess, 'TMessage> =
-    let inline ok<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = Ok(x, [])
+    let inline ok<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = Okay(x, [])
+
+    /// Wraps a value in a Success and adds a message
```
Could you please use it in a test? Thanks

pblasucci committed:
Revert last commit and improve use of non-terminal messages

+ `Result` union now back to `OK` and `Fail`
+ added `pass` helper function
+ added `Pass|Warn|Fail` active pattern
+ added narrative docs for non-terminal messages

pblasucci commented:
After thinking about thing for a bit, I reverted the `Result` union back to `OK` or `Fail`. I also added a `pass` function (basically an alias to `ok`), and an Active Pattern for looking at a result as being either:
- `Pass` ... value and no messages
- `Warn` ... value plus messages
- `Fail` ... no value, just messages

These changes keep the original code basically unchanged. However, they allow for an additional "three-track" workflow which fits _inside_ the "two-track" ROP model. I also added a help doc demonstrating the usage of the new constructs.

pblasucci committed:
corrected instances of  which should have been

mexx commented on docs/tools/templates/template.cshtml:
```diff
@@ -46,6 +46,7 @@
             
             <li class="nav-header">F# Articles</li>
             <li><a href="@Root/railway.html">Railway-oriented programming</a></li>
+            <li><a href="@Root/passawrnfail.html">Handling warnings and errors</a></li>
```
please correct the typo: should be `passwarnfail.html`

mexx commented on src/Chessie/ErrorHandling.fs:
```diff
@@ -3,11 +3,10 @@ namespace Chessie.ErrorHandling
 type Result<'TSuccess, 'TMessage> = 
     /// Represents the result of a successful computation.
-    | Ok of 'TSuccess * 'TMessage list
+    | OK of 'TSuccess * 'TMessage list
```
What does the change of case bring for the user? I wouldn't change it.

mexx commented on src/Chessie/ErrorHandling.fs:
```diff
@@ -144,8 +147,20 @@ module Combinators =
+
+    let inline failOnWarnings result =
+      match result with
+      | Warn (_,msgs) -> Fail [] |> mergeMessages msgs
```
Wouldn't `Warn (_,msgs) -> Fail msgs` bring the same result?

mexx commented on src/Chessie/ErrorHandling.fs:
```diff
@@ -18,25 +17,31 @@ type Result<'TSuccess, 'TMessage> =
+    let inline pass<'TSuccess,'TMessage> (x:'TSuccess) : Result<'TSuccess,'TMessage> = OK(x, [])
+
+    /// Wraps a value in a Success and adds a message
+    let inline warn<'TSuccess,'TMessage> (msg:'TMessage) (x:'TSuccess) : Result<'TSuccess,'TMessage> = Result.Succeed (x,msg)
```
I would implement this like other wrapper functions for consistency. `Ok(x, [ msg ])`

mexx commented:
Honestly I find such PRs really hard to review, as there are multiple things going on and the reformatting adds more struggle.
IMHO Mark Seemann (aka @ploeh) has a great [write](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/) on this.

pblasucci committed:
code clean up

pblasucci committed:
undoing still more changes

pblasucci commented:
So, I don't normally like to merge my one pull requests. But this one's been sitting out a while (and at this point I've incorporated all of the feedback), so I thought I'd go and release it. However, when I looked at the repo a bit more closely, I'm very confused. I see that GitHub Releases and NuGet have version numbers which don't correspond to the [Release Notes](https://github.com/fsprojects/Chessie/blob/master/RELEASE_NOTES.md). 

So, what's up with that?

forki commented:
Sorry I'm on vacation. Otherwise I'd have merged and released it.

Feel free to shipit. Maybe use 0.1 as version number
On Mar 19, 2015 9:52 PM, "Paulmichael Blasucci" notifications@github.com
wrote:

> So, I don't normally like to merge my one pull requests. But this one's
> been sitting out a while (and at this point I've incorporated all of the
> feedback), so I thought I'd go and release it. However, when I looked at
> the repo a bit more closely, I'm very confused. I see that GitHub Releases
> and NuGet have version numbers which don't correspond to the Release Notes
> https://github.com/fsprojects/Chessie/blob/master/RELEASE_NOTES.md.
> 
> So, what's up with that?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fsprojects/Chessie/pull/15#issuecomment-83757493.

pblasucci commented:
@forki Sorry to bother you on holiday... I'll update the version and ship it.

Thanks!
